### ISSUE TITLE
Specify module versions in cdktf.json

### ISSUE THREAD

**Issue by redeux** [hashicorp/terraform-cdk#425]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

`cdktf.json` currently pulls down the `latest` version of a module when one is specified in the config.  However, there are reasons a user would want to specify a version that isn't `latest`.

For example, the [typhoon modules'](https://registry.terraform.io/search/modules?q=typhoon) version correspond to the version of Kubernetes that it will install.

How it currently works:
```json
{
  "language": "typescript",
  "app": "npm run --silent compile && node main.js",
  "terraformProviders": [
    "google@~> 3.47.0",
    "poseidon/ct@~> 0.6.1",
    "tls@~> 2.0.0"
  ],
  "terraformModules": [
    "poseidon/kubernetes/google"
  ],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
"allowSepCharsInLogicalIds": "true"
  }
}
```

How I would expect it to work:
```json
{
  "language": "typescript",
  "app": "npm run --silent compile && node main.js",
  "terraformProviders": [
    "google@~> 3.47.0",
    "poseidon/ct@~> 0.6.1",
    "tls@~> 2.0.0"
  ],
  "terraformModules": [
    "poseidon/kubernetes/google@ 1.18.8"
  ],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
"allowSepCharsInLogicalIds": "true"
  }
}
```

But that currently results in:
```shell
➜  cdktf get
⠧ downloading and generating modules...
Error: Not Found
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/cdktf-cli/lib/get/registry-client.js:28:27)
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:315:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:634:27)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:117:17)
    at TLSSocket.socketOnData (_http_client.js:503:22)
    at TLSSocket.emit (events.js:315:20)
    at addChunk (_stream_readable.js:302:12)
    at readableAddChunk (_stream_readable.js:278:9)
    at TLSSocket.Readable.push (_stream_readable.js:217:10)
Not Found
```

### References

[Typhoon Google Cloud v1.18.8](https://registry.terraform.io/modules/poseidon/kubernetes/google/1.18.8)

**Comment by bikescholl** [comment]
Hi @redeux I was able to get this work by specifying the version in the following fashion

`"terraformModules": ["terraform-aws-modules/eks/aws@13.1.0"]`

Let me know if this works on your end.

There is currently lacking documentation on this so maybe its an opportunity for improved documentation as its not documented how to specify versions of modules in the current docs.

**Comment by redeux** [comment]
Oh interesting. It was the space between the module and the version causing it. Thanks @bikescholl 

Seems like a minor bug then, given that providers allow (and are only documented with) a space between the provider and the version.

**Comment by bikescholl** [comment]
Agreed, it tripped me up for about 20 minutes and I happened upon it by circumstance and fat fingering the module version without the space. It could be a bug potentially but I would leave that up to the cdktf team to determine.

**Comment by skorfmann** [comment]
That's resolved with #522
